ibm/config.go

```diff
@@ -594,91 +594,368 @@ func (sess clientSession) IAMNamespaceAPI() (*ns.IbmCloudFunctionsNamespaceAPIV1
 }
 
 // CIS GLB
 func (sess clientSession) CisGLBClientSession() (*cisglbv1.GlobalLoadBalancerV1, error) {
+	if sess.cisGLBErr == nil {
+		cisGLBOpt := &cisglbv1.GlobalLoadBalancerV1Options{
+			URL:            sess.cisGLBClient.Service.Options.URL,
+			Authenticator:  sess.cisGLBClient.Service.Options.Authenticator,
+			Crn:            core.StringPtr(""),
+			ZoneIdentifier: core.StringPtr(""),
+		}
+		cisGLBClient, cisGLBErr := cisglbv1.NewGlobalLoadBalancerV1(cisGLBOpt)
+		if cisGLBErr != nil {
+			return nil, fmt.Errorf(
+				"Error occured while configuring CIS GLB client: %s",
+				cisGLBErr)
+		}
+		return cisGLBClient, nil
+	}
 	return sess.cisGLBClient, sess.cisGLBErr
 }
 
 // CIS GLB Health Check/Monitor
 func (sess clientSession) CisGLBHealthCheckClientSession() (*cisglbhealthcheckv1.GlobalLoadBalancerMonitorV1, error) {
+	if sess.cisGLBHealthCheckErr == nil {
+		cisGLBHealthCheckOpt := &cisglbhealthcheckv1.GlobalLoadBalancerMonitorV1Options{
+			URL:           sess.cisGLBHealthCheckClient.Service.Options.URL,
+			Crn:           core.StringPtr(""),
+			Authenticator: sess.cisGLBHealthCheckClient.Service.Options.Authenticator,
+		}
+		cisGLBHealthCheckClient, cisGLBHealthCheckErr :=
+			cisglbhealthcheckv1.NewGlobalLoadBalancerMonitorV1(cisGLBHealthCheckOpt)
+		if cisGLBHealthCheckErr != nil {
+			return nil, fmt.Errorf(
+				"Error occured while configuring CIS GLB Health Check client: %s",
+				cisGLBHealthCheckErr)
+		}
+		return cisGLBHealthCheckClient, nil
+	}
 	return sess.cisGLBHealthCheckClient, sess.cisGLBHealthCheckErr
 }
 
 // CIS Zone Rate Limits
 func (sess clientSession) CisRLClientSession() (*cisratelimitv1.ZoneRateLimitsV1, error) {
+	if sess.cisRLErr == nil {
+		cisRLOpt := &cisratelimitv1.ZoneRateLimitsV1Options{
+			URL:            sess.cisRLClient.Service.Options.URL,
+			Crn:            core.StringPtr(""),
+			ZoneIdentifier: core.StringPtr(""),
+			Authenticator:  sess.cisRLClient.Service.Options.Authenticator,
+		}
+		cisRLClient, cisRLErr := cisratelimitv1.NewZoneRateLimitsV1(cisRLOpt)
+		if cisRLErr != nil {
+			return nil, fmt.Errorf(
+				"Error occured while cofiguring CIS Zone Rate Limit client: %s",
+				cisRLErr)
+		}
+		return cisRLClient, nil
+	}
 	return sess.cisRLClient, sess.cisRLErr
 }
 
 // CIS IP
 func (sess clientSession) CisIPClientSession() (*cisipv1.CisIpApiV1, error) {
+	if sess.cisIPErr == nil {
+		cisIPOpt := &cisipv1.CisIpApiV1Options{
+			URL:           sess.cisIPClient.Service.Options.URL,
+			Authenticator: sess.cisIPClient.Service.Options.Authenticator,
+		}
+		cisIPClient, cisIPErr := cisipv1.NewCisIpApiV1(cisIPOpt)
+		if cisIPErr != nil {
+			return nil, fmt.Errorf(
+				"Error occured while configuring CIS IP client: %s",
+				cisIPErr)
+		}
+		return cisIPClient, nil
+	}
 	return sess.cisIPClient, sess.cisIPErr
 }
 
 // CIS Page Rules
 func (sess clientSession) CisPageRuleClientSession() (*cispagerulev1.PageRuleApiV1, error) {
+	if sess.cisPageRuleErr == nil {
```

Comment:
Same here:

```go
if sess.cisPageRuleErr != nil {
	return sess.cisPageRuleClient, sess.cisPageRuleErr
}

cisPageRuleOpt := &cispagerulev1.PageRuleApiV1Options{
	URL:           sess.cisPageRuleClient.Service.Options.URL,
	Crn:           core.StringPtr(""),
	ZoneID:        core.StringPtr(""),
	Authenticator: sess.cisPageRuleClient.Service.Options.Authenticator,
}
cisPageRuleClient, cisPageRuleErr := cispagerulev1.NewPageRuleApiV1(cisPageRuleOpt)
if cisPageRuleErr != nil {
	return nil, fmt.Errorf(
		"Error occured while cofiguring CIS Page Rule client: %s",
		cisPageRuleErr)
}

return cisPageRuleClient, nil
```

Reply:
I have upgraded sdk and regenerated cis sdk. Modified the code as per your suggession.